src/platform/python/_builder.py

```diff
@@ -1,19 +1,48 @@
 import cffi
+import distutils.ccompiler
+import json
 import os, os.path
+import re
 import shlex
 import subprocess
 import sys
 
+
+def split_cmake_list(value):
+    return list(filter(None,value.split(";")))
+
+
+with open(os.environ.get("JSON_CONFIG"), "r") as f:
+    config = json.load(f)
+
+default_compiler = distutils.ccompiler.new_compiler()
+
 ffi = cffi.FFI()
 pydir = os.path.dirname(os.path.abspath(__file__))
-srcdir = os.path.join(pydir, "..", "..")
-incdir = os.path.join(pydir, "..", "..", "..", "include")
-bindir = os.environ.get("BINDIR", os.path.join(os.getcwd(), ".."))
-libdir = os.environ.get("LIBDIR")
+bindir = config["BINDIR"]
+libdirs = [config["LIBDIR"]]
+incdirs = split_cmake_list(config["INCDIRS"])
+
+cppflags = shlex.split(config["C_FLAGS"])
+cppflags += ["-I" + dir for dir in incdirs]
+cppflags += ["-D" + definition + "" for definition in split_cmake_list(config["COMPILE_DEFINITIONS"])]
+
+cpplibs = ["mgba"]
```

Comment:
`cpp` is C preprocessor. This variable name isn't correct if it refers to linker stages.